lightning/restore/restore.go

```diff
@@ -490,12 +490,17 @@ func (rc *RestoreController) restoreTables(ctx context.Context) error {
 	}
 	taskCh := make(chan task, rc.cfg.App.IndexConcurrency)
 	defer close(taskCh)
+	oriGCLifeTime, err := ObtainGCLifeTime(ctx, rc.tidbMgr.db)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	ctx2 := context.WithValue(ctx, "tikv_gc_life_time", oriGCLifeTime)
```

Comment:
https://godoc.org/context#WithValue:

> The provided key must be comparable and **should not be of type string** or any other built-in type to avoid collisions between packages using context.

Try using a global pointer like

```go
var gcLifeTimeKey struct{}

func (rc *RestoreController) restoreTables(ctx context.Context) error {
    ...
    ctx2 := context.WithValue(ctx, &gcLifeTimeKey, oriGCLifeTime)
    ...
}
```